test/bridge.go

```diff
@@ -0,0 +1,196 @@
+package test
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+)
+
+// bridgeConn is a net.Conn that represents an endpoint of the bridge.
+type bridgeConn struct {
+	br     *Bridge
+	id     int
+	readCh chan []byte
+}
+
+// Read reads data, block until the data becomes available.
+func (conn *bridgeConn) Read(b []byte) (int, error) {
+	if data, ok := <-conn.readCh; ok {
+		n := copy(b, data)
+		return n, nil
+	}
+	return 0, fmt.Errorf("bridgeConn closed")
+}
+
+// Write writes data to the bridge.
+func (conn *bridgeConn) Write(b []byte) (int, error) {
+	n := len(b)
+	conn.br.Push(b, conn.id)
+	return n, nil
```

Comment:
can you just do `return len(b), nil` just easier as the reader when I don't have to think about purpose of variables